### PR TITLE
Add readme with Express learning resources

### PR DESCRIPTION
**Pull request by PiotrBerebecki**
Include info about Express setup, routing, middleware, handlebars, testing

relates #471

### CODE REVIEW

File: coursebook/week-8/resources.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+## Resources
+
+* [What is Express and why do we use it - video tutorial](https://www.youtube.com/watch?v=45dAt9Gz8rE)
+
+* [Express hello world example](https://expressjs.com/en/starter/hello-world.html)
```
Comment (des-des) [inline coursebook/week-8/resources.md]: cool

File: coursebook/week-8/resources.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+* [Express basic routing](https://expressjs.com/en/starter/basic-routing.html)
+
+* [Express routing guide](https://expressjs.com/en/guide/routing.html)
+
+* [Express middleware](https://expressjs.com/en/guide/writing-middleware.html) - learn how middleware functions fit into application’s request-response cycle
+
+* [How to serve static files with Express](http://www.fullstacktraining.com/articles/how-to-serve-static-files-with-express)
+
+* [Express API reference](https://expressjs.com/en/4x/api.html)
```
Comment (des-des) [inline coursebook/week-8/resources.md]: Maybe put this at the top.
Most of the students may not make it that far down this list, good to try and make sure as many as possible know where the api ref is (even if it is a bit scary)

File: coursebook/week-8/resources.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+* [Express app structure - video tutorial](https://www.youtube.com/watch?v=bf8L9tQi_MQ)
+
+* [Handlebars templating engine - video tutorial](https://www.youtube.com/watch?v=1srD3Mdvf50)
+
+* [Handlebars documentation](http://handlebarsjs.com/)
+
+* [Documentation for the `express-handlebars` module](https://github.com/ericf/express-handlebars)
+
+* [Introduction to using `validator` to validate and sanitise strings](https://codeforgeek.com/2015/03/string-validation-package-nodejs/)
```
Comment (des-des) [inline coursebook/week-8/resources.md]: Just a thought:

https://www.npmjs.com/package/express-joi ?

File: coursebook/week-8/resources.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+* [How to use `express-validator` to validate form data; and `express-session` to manage user sessions](https://www.youtube.com/watch?v=hE5zeEiVqpw)
+
+* [Cookie management in Express using `cookie-parser`](https://www.codementor.io/noddy/cookie-management-in-express-js-du107rmna)
+
+* [Logging with `morgan`](https://www.npmjs.com/package/morgan)
+
+* [Test a Node RESTful API](https://scotch.io/tutorials/test-a-node-restful-api-with-mocha-and-chai)
```
Comment (des-des) [inline coursebook/week-8/resources.md]: Right now we use tape

File: coursebook/week-8/resources.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+* [Unit testing Express routers](http://evanshortiss.com/development/javascript/2016/04/15/express-testing-using-ioc.html)
+
+* [Test your REST api](https://codeforgeek.com/2015/07/unit-testing-nodejs-application-using-mocha/)
```
Comment (des-des) [inline coursebook/week-8/resources.md]: We use tape...

File: coursebook/week-8/resources.md (hunk 2 above)
Comment (PiotrBerebecki) [inline coursebook/week-8/resources.md]: Done. I've moved it as a second link just after the intro video.

File: coursebook/week-8/resources.md (hunk 3 above)
Comment (PiotrBerebecki) [inline coursebook/week-8/resources.md]: I've added this link.

File: coursebook/week-8/resources.md (hunk 4 above)
Comment (jsms90) [inline coursebook/week-8/resources.md]: @des-des how problematic do you reckon this is? If it's a good resource, I would hope that students could understand that this was a different testing framework & not be too daunted by it by week 8

But @PiotrBerebecki maybe you can find some express + tape resources instead? Might be more appropriate.

**Comment by des-des** [comment]
@PiotrBerebecki, I am not sure how you feel about this, but I am against adding a different testing framework in week 8, since we are already introducing express + handlebars..

I think using tape + supertest is good.

github.com/foundersandcoders/open-tourism-platform

https://www.codementor.io/knownasilya/testing-express-apis-with-supertest-du107mcv2
https://puigcerber.com/2015/11/27/testing-express-apis-with-tape-and-supertest/

I have not checked those but seems like lots of google hits for `tape test express`?

Thoughts?

**Comment by PiotrBerebecki** [comment]
Thanks @des-des. The workshops will include only `tape` and `supertest` for testing. I have also now updated this PR. The links in the learning resources readme include the links explaining how to test the express app with `tape` and `supertest`.

Let me know if we can improve this further before this readme can be merged.

**Pull request merged**